engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerService.java

```diff
@@ -0,0 +1,268 @@
+        String execFileName = analysis.getExecutableFileName();
+
+        log.info("Starting container: {}", containerId);
+        dockerClient.startContainerCmd(containerId).exec();
+
+        String codeDir = "/bin";
+        log.info("Copying study into container: {}", containerId);
+        dockerClient.copyArchiveToContainerCmd(containerId)
+                .withHostResource(analysisDir.getPath())
+                .withRemotePath(codeDir).exec();
+
+        String baseDir = codeDir + "/" + analysisDir.getName() + "/";
+        String workingDir = baseDir + (execFileName.split("/").length > 1 ? execFileName.split("/")[0] : "");
+        log.info("Working dir {}", workingDir);
+        String rscriptPath = baseDir + execFileName;
+        log.info("Script path {}", rscriptPath);
+        String runCommandId = dockerClient.execCreateCmd(containerId)
+                .withAttachStdout(true)
+                .withAttachStderr(true)
+                .withCmd("Rscript", rscriptPath)
+                .withWorkingDir(workingDir).exec()
+                .getId();
+
+        ResultCallback.Adapter<Frame> adapter = getAdapter(analysis, callback);
+        try {
+            log.info("Running script: {}", rscriptPath);
+            dockerClient.execStartCmd(runCommandId).exec(adapter).awaitCompletion(runtimeTimeOutSec, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            log.error("InterruptedException occurred whilst running {} in {}, message: {}", rscriptPath, containerId, e.getMessage());
+            callback.accept(Stage.EXECUTE, e.getMessage());
+        } finally {
+            stop(dockerClient, containerId);
+        }
+    }
+
+    private ResultCallback.Adapter<Frame> getAdapter(AnalysisSyncRequestDTO analysis, BiConsumer<String, String> callback) {
+        return new ResultCallback.Adapter<Frame>() {
+            @Override
+            public void onNext(Frame item) {
+                log.info(item.toString(), analysis);
+                try {
+                    String output = new String(item.getPayload());
+                    log.info("Execution output: {}", output);
+                    callback.accept(Stage.EXECUTE, output);
+                } catch (Exception e) {
+                    log.error("Error processing execution output", e);
+                    callback.accept(Stage.EXECUTE, e.getMessage());
+                }
+            }
+
+            @Override
+            public void onError(Throwable throwable) {
+                log.error("Error during execution", throwable);
+                String error = throwable.getMessage();
+                log.error("Execution error: {}", error);
+                callback.accept(Stage.EXECUTE, error);
+            }
+
+            @Override
+            public void onComplete() {
+                log.info("Execution completed");
+            }
+
+            @Override
+            public void close() throws IOException {
+                super.close();
```

Comment:
Unnecessary override.

Reply:
There are no explicit resources that need to be closed yet but I believe it would be good leave it as a placeholder for adding our own cleanup logic if necessary later.